source/databricks/settlement_report/settlement_report_job/domain/report_name_factory.py

```
@@ -0,0 +1,88 @@
+            ReportDataType.TimeSeriesQuarterly,
+        }:
+            return self._create_time_series_filename(
+                grid_area_code, energy_supplier_id, chunk_index
+            )
+        else:
+            raise NotImplementedError(
+                f"Report data type {self.report_data_type} is not supported."
+            )
+
+    def _create_time_series_filename(
+        self,
+        grid_area_code: str,
+        energy_supplier_id: str | None,
+        chunk_index: str | None,
+    ) -> str:
+
+        filename_parts = [
+            self._get_pre_fix(),
+            grid_area_code,
+            (
+                self.args.requesting_actor_id
+                if self.args.requesting_actor_market_role
+                is MarketRole.GRID_ACCESS_PROVIDER
+                else energy_supplier_id
+            ),
+            (
+                MarketRoleInFileName.GRID_ACCESS_PROVIDER
+                if energy_supplier_id is None
+                else MarketRoleInFileName.ENERGY_SUPPLIER
+            ),
+            self._get_start_date(),
+            self._get_end_date(),
+            chunk_index,
+        ]
+
+        filename_parts_without_none = [
+            part for part in filename_parts if part is not None
+        ]
+
+        return "_".join(filename_parts_without_none) + ".csv"
+
+    def _get_start_date(self) -> str:
+        time_zone_info = ZoneInfo(self.args.time_zone)
+        return self.args.period_start.astimezone(time_zone_info).strftime("%d-%m-%Y")
+
+    def _get_end_date(self) -> str:
+        time_zone_info = ZoneInfo(self.args.time_zone)
+        return (
+            self.args.period_end.astimezone(time_zone_info) - timedelta(days=1)
+        ).strftime("%d-%m-%Y")
+
+    def _get_pre_fix(self) -> str:
+        if self.report_data_type == ReportDataType.TimeSeriesHourly:
+            return "TSSD60"
+        elif self.report_data_type == ReportDataType.TimeSeriesQuarterly:
+            return "TSSD15"
```

Should we add a final "else"?

---

there is a guard in the create method. We can also raise an exception here if you prefer